### PR TITLE
Handle unregistered resources

### PR DESCRIPTION
**Pull request by janisz**
## Description

In #961 we added a check for most of the resources. When resource was unknown we silently skip it. This PR introduces a panic when we are unable to get resource metadata.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -269,7 +269,7 @@ clean-easyjson-srcs:
 .PHONY: go-generated-srcs
 go-generated-srcs: deps clean-easyjson-srcs go-easyjson-srcs $(MOCKGEN_BIN) $(STRINGER_BIN) $(GENNY_BIN)
 	@echo "+ $@"
-	PATH="$(PATH):$(BASE_DIR)/tools/generate-helpers" go generate -v -x ./...
+	PATH="$(PATH):$(BASE_DIR)/tools/generate-helpers" ROX_VERIFY_IMAGE_SIGNATURE=true go generate -v -x ./...
```
Comment (connorgorman) [inline Makefile:272]: Why turn the flag on here?

**Review by connorgorman — APPROVED** [review]
Curious about the flag, but otherwise looks good

File: Makefile (hunk 1 above)
Comment (janisz) [reply Makefile:272]: Good question. The resource is behind feature flag so when I try to generate code for it I cannot get it's metadata unless I set this flag. I'm not sure how to solve this issue. Another way will be to have a list of all resources (including those behind feature flag).

Comment (janisz) [reply Makefile:272]: Maybe having a method `resources.ListAllDisabledMetadata()` will be cleaner and future proof

**Comment by janisz** [comment]
Check generated files fails because in go < 1.18 conditions in templates are evaluated eagerly and then compared this is fixed by https://github.com/golang/go/issues/31103 so we need a workaround for now

**Pull request merged**